pallets/loans/src/entities/pricing/external.rs

```diff
@@ -152,17 +185,25 @@ impl<T: Config> ExternalActivePricing<T> {
 		Ok(debt.ensure_sub(outstanding_notional)?)
 	}
 
-	pub fn present_value(&self, pool_id: T::PoolId) -> Result<T::Balance, DispatchError> {
-		self.outstanding_principal(pool_id)
+	pub fn present_value(
+		&self,
+		pool_id: T::PoolId,
+		maturity: Seconds,
+	) -> Result<T::Balance, DispatchError> {
+		self.outstanding_principal(pool_id, maturity)
 	}
 
-	pub fn present_value_cached<Prices>(&self, cache: &Prices) -> Result<T::Balance, DispatchError>
+	pub fn present_value_cached<Prices>(
+		&self,
+		cache: &Prices,
+		maturity: Seconds,
+	) -> Result<T::Balance, DispatchError>
 	where
 		Prices: DataCollection<T::PriceId, Data = PriceOf<T>>,
 	{
 		let price = match cache.get(&self.info.price_id) {
 			Ok(data) => data.0,
-			Err(_) => self.latest_settlement_price,
+			Err(_) => self.linear_accrual_price(maturity)?,
 		};
 		Ok(self.outstanding_quantity.ensure_mul_int(price)?)
 	}
```

Comment:
...the externally priced loan will fail to obtain the "real value" and will generate the missing price for itself using the linear accrual method.

Later, faking or not, the `update_portfolio()` with a list of prices is independent of this process.

Reply:
But this way we can not fake the oracle prices, in cases where we want to simulate the nav execution with new oracles. And this would mean closing an epoch is not protected if the oracle is outdated.

Reply:
> But this way we can not fake the oracle prices

We can because when we fake them, the value will come from the `Ok()` branch (line 205).

Reply:
Well, it affects. Because now, when there are no oracles, the price is the linear accrual instead of `settlement_price`. Ideally, after this change, we should do a `portfolio_update()` to consolidate the changes in the present value affected by this.

We could even add it as part of the migration. WDYT @mustermeiszer?

Reply:
Not sure if it is for PoolFees. But it can be for `pallet-loans` because we are modifying how the present_value is computed but we are not updating the portfolio with this new computation

Reply:
> But it can be for pallet-loans because we are modifying how the present_value() is computed but we are not updating the portfolio with this new computation

Finally added. If finally you think we shouldn't, I can revert it. Commit: [b760ad6](https://github.com/centrifuge/centrifuge-chain/pull/1739/commits/b760ad6c1fcea264a11e206d3b7d66018a01f257)